engine/src/main/java/org/hibernate/validator/internal/properties/javabean/JavaBeanHelper.java

```diff
@@ -0,0 +1,162 @@
+		this.getterPropertySelectionStrategy = getterPropertySelectionStrategy;
+	}
+
+	public Optional<JavaBeanField> findDeclaredFieldProperty(Class<?> declaringClass, String property) {
+		Contracts.assertNotNull( declaringClass, MESSAGES.classCannotBeNull() );
+
+		Field field = run( GetDeclaredField.action( declaringClass, property ) );
+		return Optional.ofNullable( field ).map( JavaBeanField::new );
+	}
+
+	public Optional<JavaBeanGetter> findDeclaredGetterProperty(Class<?> declaringClass, String property) {
+		Contracts.assertNotNull( declaringClass, MESSAGES.classCannotBeNull() );
+		for ( String possibleName : getterPropertySelectionStrategy.getPossibleMethodNamesForGetterProperty( property ) ) {
+			Method getter = run( GetDeclaredMethod.action( declaringClass, possibleName ) );
+			if ( getter != null ) {
+				return Optional.of( new JavaBeanGetter( declaringClass, getter, getterPropertySelectionStrategy.getPropertyName( new JavaBeanConstrainableExecutable( getter ) ) ) );
+			}
+		}
+		return Optional.empty();
+	}
+
+	public Optional<JavaBeanGetter> findGetterProperty(Class<?> declaringClass, String property) {
+		Contracts.assertNotNull( declaringClass, MESSAGES.classCannotBeNull() );
+
+		Optional<JavaBeanGetter> declaredGetterProperty = findDeclaredGetterProperty( declaringClass, property );
+
+		if ( declaredGetterProperty.isPresent() ) {
+			return declaredGetterProperty;
+		}
+		else {
+			// look for getters from the superclasses and interfaces
+			for ( String possibleName : getterPropertySelectionStrategy.getPossibleMethodNamesForGetterProperty( property ) ) {
+				Method getter = run( GetMethod.action( declaringClass, possibleName ) );
+				if ( getter != null ) {
+					return Optional.of( new JavaBeanGetter( declaringClass, getter, getterPropertySelectionStrategy.getPropertyName( new JavaBeanConstrainableExecutable( getter ) ) ) );
+				}
+			}
+			return Optional.empty();
+		}
+	}
+
+	public Optional<JavaBeanMethod> findDeclaredMethodByNameAndParameters(Class<?> declaringClass, String methodName, Class<?>... parameterTypes) {
+		Method method = run( GetDeclaredMethod.action( declaringClass, methodName, parameterTypes ) );
+		if ( method == null ) {
+			return Optional.empty();
+		}
+		else {
+			return Optional.of( executable( declaringClass, method ) );
+		}
+	}
+
+	public <T> Optional<JavaBeanConstructor> findDeclaredConstructorByParameters(Class<T> declaringClass, Class<?>... parameterTypes) {
+		Constructor<T> constructor = run( GetDeclaredConstructor.action( declaringClass, parameterTypes ) );
+		if ( constructor == null ) {
+			return Optional.empty();
+		}
+		else {
+			return Optional.of( new JavaBeanConstructor( constructor ) );
+		}
+	}
+
+	public JavaBeanExecutable<?> executable(Executable executable) {
+		if ( executable instanceof Constructor ) {
+			return new JavaBeanConstructor( (Constructor<?>) executable );
+		}
+
+		return executable( executable.getDeclaringClass(), ( (Method) executable ) );
```

Comment:
Can't you just call `executable(Class<?> declaringClass, Executable executable)` and remove the `if()`?